### PR TITLE
feat: refactored devices command and added health and status flags.

### PR DESCRIPTION
**Pull request by bflorian**
Refactored devices command to move utility functions into a separate module and added flags to include device health and device status information to device list and get commands.

## Checklist

- [x] I have read the **[CONTRIBUTING](../CONTRIBUTING.md)** document
- [x] Any required documentation has been added
- [x] My code follows the code style of this project (`npm run lint` produces no warnings/errors)
- [x] I have added tests to cover my changes

### CODE REVIEW

File: packages/cli/src/commands/devices.ts [hunk 1]
```diff
@@ -84,7 +108,16 @@ export default class DevicesCommand extends APICommand<typeof DevicesCommand.fla
 				}
 				return devices
 			},
-			id => this.client.devices.get(id),
+			async (id) => {
+				const device = await this.client.devices.get(id, deviceGetOptions)
+				// Note -- we have to do this explicitly because the API does not honor the includeHealth parameter
+				// for individual devices
+				if (this.flags.health) {
```
Comment (john-u) [inline packages/cli/src/commands/devices.ts:115]: This could be refactored to be a parallel call since we aren't using the device to get the health state.

Comment (bflorian) [reply packages/cli/src/commands/devices.ts:114]: Done

File: packages/cli/src/lib/commands/devices-util.ts [hunk 2]
```diff
@@ -1,27 +1,102 @@
...
+				}
+			}
+			output += table.toString()
+			output += '\n'
+		}
+	}
+	return output
+}
+
+export const buildEmbeddedStatusTableOutput = (tableGenerator: TableGenerator, data: Device): string => {
```
Comment (rossiam) [inline packages/cli/src/lib/commands/devices-util.ts:44]: I might play with trying to merge these two functions after you merge this if that's okay. (I'm thinking maybe a single generic function and then two glue functions that call it, assuming I can figure out how to make the types work.)

**Review by rossiam — APPROVED** [review]

File: packages/cli/src/__tests__/lib/commands/devices-util.test.ts [hunk 3]
```diff
@@ -1,11 +1,203 @@
...
+			push: tablePushMock,
+			toString: tableToStringMock,
+		} as unknown as Table
+		const newOutputTableMock = jest.fn().mockReturnValue(tableMock)
+
+		const tableGeneratorMock: TableGenerator = {
+			newOutputTable: newOutputTableMock,
+			buildTableFromItem: jest.fn(),
+			buildTableFromList: jest.fn(),
+		} as TableGenerator
```
[the remark this answers is not in the document]
Comment (bflorian) [reply packages/cli/src/__tests__/lib/commands/devices-util.test.ts:58]: Will refactor these in next PR, which is in progress

**Pull request merged**